WalletWasabi/WabiSabi/Client/CoinRefrigerator.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using WalletWasabi.Blockchain.TransactionOutputs;
+
+namespace WalletWasabi.WabiSabi.Client
+{
+	public class CoinRefrigerator
+	{
+		private Dictionary<SmartCoin, DateTimeOffset> CoinsInFridge { get; } = new();
+		private TimeSpan FreezeTime { get; } = TimeSpan.FromSeconds(90);
+
+		public void Freeze(IEnumerable<SmartCoin> coins)
+		{
+			foreach (var coin in coins)
+			{
+				if (!CoinsInFridge.ContainsKey(coin))
+				{
+					CoinsInFridge.TryAdd(coin, DateTimeOffset.UtcNow);
+				}
+				else
+				{
+					CoinsInFridge[coin] = DateTimeOffset.UtcNow;
+				}
+			}
+		}
+
+		public bool IsFrozen(SmartCoin coin)
+		{
+			if (!CoinsInFridge.TryGetValue(coin, out var starTime))
+			{
+				return false;
+			}
+
+			if (starTime.Add(FreezeTime) > DateTimeOffset.UtcNow)
```

Comment:
At this point, it's not really important but: If you set in `Freeze(Coins)` using `CoinsInFridge[coin] = DateTimeOffset.UtcNow.Add(FreezeTime);` then you don't need to use `.Add(FreezeTime)` on this line and as a benefit, you can freeze your coins at variable times if useful in the future. 

(Also might save some unnecessary computations)

Reply:
That's how i implemented in my toy PR but this is also okay.